main/integration-tests/__test__/api-tests/workspace-service-catalogs/create-workspace-service-catalog.test.js

```diff
@@ -166,4 +179,90 @@ describe('Create workspace-service-catalog scenarios', () => {
       });
     });
   });
+  describe('Creating a workspace service catalog environment with studies', () => {
+    it('for EC2Linux', async () => {
+      // const researcher1Session = await setup.createResearcherSession();
+      // Things to do
+      // 1. Create user
+      const admin1Session = await setup.createAdminSession();
+      const username = setup.gen.username();
+      const defaultUser = admin1Session.resources.users.defaults({ username });
+      await expect(admin1Session.resources.users.create(defaultUser)).resolves.toMatchObject({
+        username,
+      });
+      // 2. Assign Project ID to user (assigned by default)
+
+      // 3. Create My Study, Org Study, Data Source study
+      const studyIds = [];
+      let studyId = setup.gen.string({ prefix: `create-study-ray-my-study` });
+      await expect(
+        admin1Session.resources.studies.create({ id: studyId, name: studyId, category: 'My Studies' }),
+      ).resolves.toEqual(
+        expect.objectContaining({
+          id: studyId,
+        }),
+      );
+      studyIds.push(studyId);
+
+      studyId = setup.gen.string({ prefix: `create-study-ray-org-study` });
+      await expect(
+        admin1Session.resources.studies.create({ id: studyId, name: studyId, category: 'Organization' }),
+      ).resolves.toEqual(
+        expect.objectContaining({
+          id: studyId,
+        }),
+      );
+      studyIds.push(studyId);
+
+      // 4. Create workspace with the above studies
+      const workspaceName = setup.gen.string({ prefix: 'workspace-sc-test' });
+
+      const env = await admin1Session.resources.workspaceServiceCatalogs.create({
+        name: workspaceName,
+        envTypeId: setup.defaults.envTypes.ec2Linux.envTypeId,
+        envTypeConfigId: setup.defaults.envTypes.ec2Linux.envTypeConfigId,
+        studyIds,
+        description: 'assignment',
+        projectId: setup.defaults.project.id,
+        cidr: '54.240.196.186/32',
+      });
+      expect(env).toEqual(
+        expect.objectContaining({
+          name: workspaceName,
+          envTypeId: setup.defaults.envTypes.ec2Linux.envTypeId,
+          envTypeConfigId: setup.defaults.envTypes.ec2Linux.envTypeConfigId,
+          studyIds,
+        }),
+      );
+
+      const workflows = await admin1Session.resources.workflows
+        .versions('wf-provision-environment-sc')
+        .version(1)
+        .instances()
+        .get();
+      let result;
+      for (const w of workflows) {
```

Comment:
Do we have a helper method to do this already? Should be able to find the workflow without iterating over all of them. If not, I think lodash has a _.get() method that can do this as well in a slightly cleaner way.

Reply:
I added the _.filter method from lodash to simplify this piece of code.